src/Build.UnitTests/Graph/GraphTestingUtilities.cs

```diff
@@ -205,5 +215,32 @@ internal static string GetProjectPath(ProjectGraphNode node)
                 defaultTargets ?? "Build",
                 extraContent);
         }
+
+        internal static IEnumerable<ProjectGraphNode> ComputeClosure(ProjectGraphNode node)
```

Comment:
This doesn't seem to have a way to detect cyclical references, and there could easily be two projects that reference a single project, thus adding all its references etc. twice. Maybe add an "I've seen this before" escape?

Reply:
This is a test utility I use to assert transitive reference test results, not the real thing, so I'd rather keep it simple. Also, cycles are not allowed in static graph, so I don't concern myself with them.